app/src/main/java/com/willowtree/vocable/presets/AddToCategoryPickerFragment.kt

```diff
@@ -0,0 +1,162 @@
+            requireActivity(),
+            BaseViewModelFactory()
+        ).get(AddToCategoryPickerViewModel::class.java)
+
+        subscribeToViewModel()
+
+        with(addToCategoryPickerViewModel) {
+            buildCategoryList(phraseText)
+        }
+    }
+
+    private fun subscribeToViewModel() {
+        addToCategoryPickerViewModel.categoryMap.observe(viewLifecycleOwner, Observer {
+            it?.let { map ->
+                with(ArrayList(map.keys)) {
+                    val categoriesExist = this.size != 0
+
+                    binding.emptyStateTitle.isVisible = !categoriesExist
+                    binding.emptyStateText.isVisible = !categoriesExist
+                    binding.categoryPagerBackButton.isEnabled = categoriesExist
+                    binding.categoryPagerForwardButton.isEnabled = categoriesExist
+
+                    // empty state
+                    if (this.size == 0) {
+                        binding.categoryPageNumber.text = getString(R.string.phrases_page_number, 1, 1)
+                    } else {
+                        binding.categoryHolder.apply {
+                            isSaveEnabled = false
+                            adapter = categoriesAdapter
+                            categoriesAdapter.setItems(this@with)
+
+                            // Move adapter to middle so user can scroll both directions
+                            val middle = categoriesAdapter.itemCount / 2
+                            if (middle % categoriesAdapter.numPages == 0) {
+                                setCurrentItem(middle, false)
+                            } else {
+                                val mod = middle % categoriesAdapter.numPages
+                                setCurrentItem(
+                                    middle + (categoriesAdapter.numPages - mod),
+                                    false
+                                )
+                            }
+                        }
+                    }
+                }
+            }
+        })
+    }
+
+    override fun getAllViews(): List<View> {
+        return emptyList()
+    }
+
+    inner class CategoriesPagerAdapter(fm: FragmentManager) :
+            VocableFragmentStateAdapter<Category>(fm, viewLifecycleOwner.lifecycle) {
+        override fun setItems(items: List<Category>) {
+            super.setItems(items)
+            setPagingButtonsEnabled(numPages > 1)
+        }
+
+        private fun setPagingButtonsEnabled(enable: Boolean) {
+            with (binding) {
+                categoryPagerBackButton.isEnabled = enable
+                categoryPagerForwardButton.isEnabled = enable
+            }
+        }
+
+        override fun getMaxItemsPerPage(): Int = maxCategories
+
+        override fun createFragment(position: Int): Fragment {
+            val categories = getItemsByPosition(position)
+
+            return AddToCategoryPickerListFragment.newInstance(args.phraseText, categories)
```

Comment:
Is `args.phraseText` ever null? If it is, is that handled?

Reply:
It's a non-nullable argument to the Navigation component, so I don't think it can be null, or it wouldn't even get to this screen.